### PR TITLE
*: generalize and link to the external storage docs from Lightning

### PR DESCRIPTION
**Pull request by kennytm**
### What is changed, added or deleted? (Required)

Fix pingcap/docs#4445.

Generalize the "BR storage" page to "external storage", and talk about all associated migration services.

Link back to this page from their documents.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs/issues/4555

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/backup-and-restore-storages.md [hunk 1]
```diff
@@ -46,12 +56,12 @@ S3 和 GCS 等云存储有时需要额外的连接配置，你可以为这类配
 
 > **注意：**
 >
-> 不建议在存储 URL 中直接传递访问密钥和 secret 访问密钥，因为这些密钥是明文记录的。BR 尝试按照以下顺序从环境中推断这些密钥：
+> 不建议在存储 URL 中直接传递访问密钥和 secret 访问密钥，因为这些密钥是明文记录的。迁移工具尝试按照以下顺序从环境中推断这些密钥：
```
Comment (glorv) [inline br/backup-and-restore-storages.md:59]: BR 和 dumpling 都支持通过命令行参数 --s3.* 设置 S3 相关的参数，但是 lightning 目前只支持在 URL 里面设置，这块是不是也需要描述一下..

Comment (kennytm) [reply br/backup-and-restore-storages.md:59]: @glorv PTAL

File: br/backup-and-restore-storages.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
 ---
-title: BR 存储
-summary: 了解 BR 中所用存储服务的 URL 格式。
+title: 远端存储
```
Comment (overvenus) [inline br/backup-and-restore-storages.md:2]: What about "外部存储"？

Cc @IANTHEREAL

Comment (kennytm) [reply br/backup-and-restore-storages.md:2]: this is the terminology in #3925. but this was translated from pingcap/docs#2618 "remote destination" not "storage" 🤔

File: br/backup-and-restore-storages.md (hunk 1 above)
Comment (glorv) [reply br/backup-and-restore-storages.md:59]: LGTM

**Comment by glorv** [comment]
LGTM

**Comment by overvenus** [comment]
LGTM

File: br/backup-and-restore-storages.md [hunk 3]
```diff
@@ -1,16 +1,16 @@
 ---
-title: BR 存储
-summary: 了解 BR 中所用存储服务的 URL 格式。
+title: 外部存储
```
Comment (TomShawn) [inline br/backup-and-restore-storages.md:2]: + `外部存储` is not only for BR. So we need to change the file name (like `tools-remote-storage.md`) and put the file out of BR directory.
+ We also need to move its position in the TOC menu. @kissmydb @overvenus @IANTHEREAL Any good idea?

Comment (kennytm) [reply br/backup-and-restore-storages.md:2]: For tools there's a bunch of these "shared language" description needing such position. [Table-filter](https://docs.pingcap.com/zh/tidb/stable/table-filter) is another instance.

Comment (TomShawn) [reply br/backup-and-restore-storages.md:2]: + But in this case, the BR-specific file name is clearly not suitable and can be misleading.
+ For the file position, what about `/tools-external-storage.md`?
+ For TOC, what about 
image?
+ For the table filter case, put it under `TiDB 生态工具` LGTM. But it also needs opinion from others.

Comment (overvenus) [reply br/backup-and-restore-storages.md:2]: It belongs to BR, so `br/backup-and-restore-storages.md` LGTM.

Comment (IANTHEREAL) [reply br/backup-and-restore-storages.md:2]: I agree with @TomShawn , Thanks for the user's understanding perspective

Comment (TomShawn) [reply br/backup-and-restore-storages.md:2]: @kennytm PTAL

Comment (kennytm) [reply br/backup-and-restore-storages.md:2]: Note: we have discussed about this internally, and the consensus is we will perform the rename in a separate PR.

**Comment by IANTHEREAL** [comment]
LGTM

File: br/backup-and-restore-storages.md [hunk 4]
```diff
@@ -19,19 +19,29 @@ BR 支持以下存储服务：
...
+## URL 参数
 
 S3 和 GCS 等云存储有时需要额外的连接配置，你可以为这类配置指定参数。例如：
 
 {{< copyable "shell-regular" >}}
 
-```shell
-./br backup full -u 127.0.0.1:2379 -s 's3://bucket-name/prefix?region=us-west-2'
+```bash
+# 用 Dumpling 导出数据到 S3
```
Comment (TomShawn) [inline br/backup-and-restore-storages.md:29]: Suggest writing like this for readability:

+ 用 Dumpling 导出数据到 S3：

    {{< copyable "shell-regular" >}}
    
    ```bash
    ./dumpling -u root -h 127.0.0.1 -P 3306 -B mydb -F 256MiB \
        -o 's3://my-bucket/sql-backup?region=us-west-2'
    ```

+ 用 TiDB Lightning 从 S3 导入数据：

   {{< copyable "shell-regular" >}}
    
    ```bash
    ./tidb-lightning --tidb-port=4000 --pd-urls=127.0.0.1:2379 --backend=local --sorted-kv-dir=/tmp/sorted-kvs \
        -d 's3://my-bucket/sql-backup?region=us-west-2'
    ```

File: br/backup-and-restore-storages.md [hunk 5]
```diff
@@ -1,16 +1,16 @@
 ---
-title: BR 存储
-summary: 了解 BR 中所用存储服务的 URL 格式。
+title: 外部存储
+summary: 了解 BR、Lightning 和 Dumpling 中所用存储服务的 URL 格式。
```
Comment (TomShawn) [inline br/backup-and-restore-storages.md:3]: Global comment: Lighting --> TiDB Lightning

**Comment by kennytm** [comment]
PTAL @TomShawn

**Comment by TomShawn** [comment]
Follow-up: Change the file name for `外部存储` and remove it out of `/br`. @IANTHEREAL will work on its new position in TOC.

**Pull request merged**